src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -3663,8 +3672,29 @@ public override BoundNode VisitDefaultExpression(BoundDefaultExpression node)
 
         public override BoundNode VisitIsOperator(BoundIsOperator node)
         {
+            Debug.Assert(!this.IsConditionalState);
+
+            int slot = -1;
+            var operand = node.Operand;
+            if (operand.Type.IsReferenceType)
```

Comment:
Please test unconstrained generic types.
```
static void F<T>(T t, object? o)
{
    if (t is string) t.ToString();
    if (o is T) o.ToString();
}
```

Reply:
Thanks! That's caught two issues:
- I've adjusted the check here (we could even remove it, but why track variables with struct type?)
- even though we realize that `t` isn't null after `t is string`, we still report a possible null de-reference. That's an existing issue with variables of unconstrained generic type (added a PROTOTYPE).